itou/www/employee_record_views/forms.py

```diff
@@ -144,19 +145,18 @@ def __init__(self, *args, **kwargs):
         self.fields["hexa_lane_name"].validators = [address_re_validator]
         self.fields["hexa_additional_address"].validators = [address_re_validator]
 
+    def clean_hexa_post_code(self):
+        post_code = self.cleaned_data["hexa_post_code"]
+        if not City.objects.filter(post_codes__contains=[post_code]).exists():
+            raise forms.ValidationError("Le code postal ne correspond à aucune ville.")
+        return post_code
+
     def clean(self):
         super().clean()
 
         if self.cleaned_data.get("hexa_std_extension") and not self.cleaned_data.get("hexa_lane_number"):
             raise forms.ValidationError("L'extension doit être saisie avec un numéro de voie")
 
-        hexa_commune = self.cleaned_data.get("hexa_commune")
-        post_code = self.cleaned_data.get("hexa_post_code")
-
-        # Check basic coherence between post-code and INSEE code:
-        if post_code and hexa_commune and post_code[:2] != hexa_commune.code[:2]:
```

Comment:
Mais est-ce qu'on ne voudrait pas, si "hexa_commune.city is not None`, vérifier que le `post_code` est bien dans `hexa_commune.city.post_codes` ?

Reply:
Bon ça aurait du sens si le référentiel ASP et la liste des villes étaient mieux synchronisé (moins de 500 concernées)
Ce n'est actuellement pas le cas donc pas la peine de s'embêter.